libs/index/odc/index/stac.py

```diff
@@ -166,7 +172,7 @@ def _get_stac_properties_lineage(input_stac: Document) -> Tuple[Document, Any]:
         MAPPING_STAC_TO_EO3.get(key, key): _convert_value_to_eo3_type(key, val)
         for key, val in properties.items()
     }
-    if prop.get("odc:processing_datetime") is None:
+    if prop.get("odc:processing_datetime") is None and properties["datetime"] is not None:
```

Comment:
this will raise `KeyError` if `datetime` field is not populated

Reply:
Thanks for fixing this @alexgleith. FWIW I believe STAC _requires_ a datetime field, which is then allowed to be null... good to not crash out here anyway though.